### PR TITLE
Switch AQI calculations to use the EPA recommended corrections for PA Sensors.

### PR DESCRIPTION
**Pull request by obra**
Please check this work CAREFULLY. 

As of this PR, we are -not- yet averaging across at least two minutes of readings and throwing a fit if values are off by more than 5% as recommended by them.

Sources:

https://thebolditalic.com/understanding-purpleair-vs-airnow-gov-measurements-of-wood-smoke-pollution-562923a55226
https://cfpub.epa.gov/si/si_public_record_report.cfm?dirEntryId=349513&Lab=CEMM&simplesearch=0&showcriteria=2&sortby=pubDate&timstype=&datebeginpublishedpresented=08/25/2018

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -228,6 +230,11 @@
     return aqiFromPM(0.778 * pm + 2.65);
   }
 
+  function epaAQIFromPMandHumidity(pm, humidity) {
+    return aqiFromPM((0.543*pm) - (0.0844*humidity) + 5.604);
```
Comment (skalnik) [inline app.js]: Hm, from earlier in the [The EPA's PDF](https://cfpub.epa.gov/si/si_public_record_report.cfm?dirEntryId=349513&Lab=CEMM&simplesearch=0&showcriteria=2&sortby=pubDate&timstype=&datebeginpublishedpresented=08/25/2018), it seems like this should be `(0.52 * pm) -(0.085 * humidity) +5.71`, though I did find your formula at the end of the slides. However, [The Bold Italic's medium post](https://thebolditalic.com/understanding-purpleair-vs-airnow-gov-measurements-of-wood-smoke-pollution-562923a55226) seems to suggest `(0.524 * pm) - (0.0852 * humidity) + 5.71`. 

The first is also cited in https://fire.airnow.gov's FAQ. I wonder what' up with these differences in the formulas presented.

Also, at the end of the EPA's slide deck, they say we should be averaging the values over the last hour. Do you think this is still valuable if we don't do that? fire.airnow.gov's description seems to not talk about averaging over the last hour, so maybe that tweaked calculation would be better? I'm not really sure.

File: app.js [hunk 2]
```diff
@@ -85,17 +85,19 @@
 
   function updateAQI(sensor) {
     let pm25s = [];
+    let humidity = sensor.results[0].humidity;
```
Comment (skalnik) [inline app.js:93]: Looking at the final slide of the EPA deck, it seems like we'll need to average the humidity over some time if we want to do it as implemented on that slide.

**Review by skalnik — COMMENTED** [review]
Seems like a reasonable start! I'm kinda hesitant on implementing this without any way to verify the calculation. The numbers don't line up with what I see on https://fire.airnow.gov/, but that's somewhat to be expected due to the difference in adjustment calculation we're doing.

Additionally, overall the recommendation from the cited article is still AQandU, despite it not being perfect, so that may be another reason to stick with that value, even with the pit falls.

**Comment by obra** [comment]
I would agree pretty strongly with a decision to leave this hanging out for a while until we can see another implementation.
I ....sort of want to stick all the alternate AQI calculations on the page in invisible text, just so we can watch them. But I don't feel strongly about it.

File: app.js (hunk 1 above)
Comment (obra) [reply app.js]: The Bold Italic's post is clearly drawing on the EPA PDF and is pretty clear that they don't really understand the math.

fire.airnow.gov's numbers appear to be updated only once an hour at best.

Understanding whether we need to be doing averaging over time for this technique to be valid and why there are two different versions of the formula feels...important.   

I think one of us ought to reach out to the folks who authored that EPA document to ask these questions. The contact address they gave us is CEMM@epa.gov. You want me to do it & cc you or you wanna take it?

**Comment by skalnik** [comment]
What about a second, unlinked, page with all the values on it?

File: app.js (hunk 1 above)
Comment (obra) [reply app.js]: actually, there are better contacts in the pdf itself. I'm writing a draft, but happy to run it by you or let you run with it.

Comment (skalnik) [reply app.js]: I'm happy to review it, but feel free to send it! I mean to be a bottleneck or anything.

Comment (obra) [reply app.js]: Dropped you a DM on twitter

**Comment by skalnik** [comment]
I done any comparisons or anything but I just noticed that PurpleAir now includes an EPA adjustment!

**Comment by daylen** [comment]
As mentioned above, now that PurpleAir has an EPA adjustment, we could verify that this PR matches and then merge it in.

**Comment by skalnik** [comment]
I adjusted this to use the values posted on the PurpleAir website and fixed the link to send the EPA, and the values seem to align for me (seen them at most 1 off, I figure due to timing). If anyone else wants to try this out and make sure it seems good that'd be great!

**Comment by skalnik** [comment]
Hm, checked on my bike ride this morning and something seems off!

IMG_5760

**Comment by skalnik** [comment]
Hm, that URL may have been outdated. I think Vercel made some changes. https://aqi-wtf-izgia4xy5.vercel.app seems to not include the `.` at the first line, but the URL I was going to did, so maybe this is fine!?

**Comment by kylebshr** [comment]
👋 Any update on this? Hoping to borrow your implementation but wanted to make sure it all checks out

File: app.js [hunk 3]
```diff
@@ -90,17 +90,19 @@
...
 
     announceState("Calculating AQI");
 
     for (const subsensor of sensor.results) {
       if (!bustedSensor(subsensor)) {
-        pm25s.push(parseFloat(subsensor["PM2_5Value"]));
+        pm25s.push(parseFloat(subsensor["pm2_5_cf_1"]));
```
Comment (kylebshr) [inline app.js:99]: What does this change? And why? PurpleAir docs are very light 😅

**Comment by obra** [comment]
CF1 is “ correction factor 1” and that change is because it is  what is specified in the EPA presentation about how to calculate this number

**Comment by kylebshr** [comment]
Awesome, thanks! Copied this over to my app and spot-checking against purple air real-time is looking correct. Going to ship it I think!

**Comment by skalnik** [comment]
Yeah I've been checking this for the past couple of weeks and I haven't gotten a weird number since and it always lines up with the Purple Air numbers!

**Pull request merged**